### PR TITLE
Demo js

### PR DESCRIPTION
**Pull request by zach-planet-nine**
This PR updates the client-ios demo, and the javascript a typescript servers to handle associating keys, and calling APIs on each other's servers. A video of this demo can be found here:

https://www.planetnineapp.com/sessionless-demo-1

### CODE REVIEW

**Review by FssAy — CHANGES_REQUESTED** [review]
The README wasn't updated even tho new endpoints have been added. I think it could be a good idea to include them in addition with the link to the video (https://www.planetnineapp.com/sessionless-demo-1)

File: src/javascript/example/server/src/demo/demo.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      timestamp: payload.timestamp1
+    });
+    
+    const message2 = JSON.stringify({
+      uuid: payload.uuid2,
+      timestamp: payload.timestamp2,
+      pubKey: payload.pubKey
+    });
+
+    if(!sessionless.verifySignature(signature1, message1, user.pubKey) ||
```
Comment (FssAy) [inline src/javascript/example/server/src/demo/demo.js:28]: There was a discussion of constructing messages to verify a signature. The message used to create the signature cannot differ from the one used to verify it. 
If we construct a message from the payload like in the line `17` and `22` the `verifySignature` function might fail depending of the JSON implementation. 
One end can encode the message as:
```json
{"uuid": "<val>", "timestamp": "<val>"}
```
and the second one as:
```json
{"timestamp": "<val>", "uuid": "<val>"}
```
The content is the same, but the message is different. I think we should highly avoid that.

The current body of this request looks like this:
```json
{
   "uuid1": "...",
   "timestamp1": "...",
   "signature1": "...",
   "uuid2": "...",
   "timestamp2": "...",
   "signature2": "...",
   "pubKey": "..."
}
```
We could encode each message into Base64 and then use the encoded string to create the signature. In practice that's how the body would look like now:
```json
{
   "message1": "eyJ1dWlkIjogInZhbDEiLCAidGltZXN0YW1wIjogInZhbDEifQ==",
   "signature1": "...",
   "message2": "eyJ1dWlkIjogInZhbDIiLCAidGltZXN0YW1wIjogInZhbDIifQ==",
   "signature2": "..."
   "pubKey": "..."
}
```
Where each `"message"` is encoded JSON: `{"uuid": "valX", "timestamp": "valX"}`.

Now it's possible to easily get the data and verify the signature.

The signing process could look like this:
```js
message = {
   uuid: "...",
   timestamp: "...",
};

messageEncoded = base64(message);

signature = sign(messageEncoded);

payload = {
   message: messageEncoded,
   signature: signature,
}
```

**Review by FssAy — CHANGES_REQUESTED** [review]
I've found a major issue in the code while updating the Rust implementation based on this demo (see bellow).

**Review by FssAy — APPROVED** [review]
Great! Now it should be easier to understand for anyone looking to contribute.

**Review by habomane — APPROVED** [review]
Looks good!

**Pull request merged**